### ISSUE TITLE
Validate Mixed Clusters (windows + linux) for Azure US Gov cloud 

### ISSUE THREAD

**Issue by rosskirkpat** [rancher/rancher#25406]
**What kind of request is this (question/bug/enhancement/feature request):**
feature request

**Steps to reproduce (least amount of steps as possible):**

When using a non-default Azure cloud (for example US Gov), creating a Windows cluster fails due to a mismatch in the 'cloud' argument between the RKE-tools and the kubelet. The Azure Cloud provider states the valid values for the 'cloud' parameter (https://github.com/kubernetes-sigs/cloud-provider-azure/blob/master/docs/cloud-provider-config.md) but RKE-tools attempts to set the cloud name directly based on the cloud setting - which is incompatible for non-default cloud settings.

For example, the azure cloud provider states 'AzureUSGovernmentCloud' is a valid entry, whilst the command `az cloud show` shows the cloud name is AzureUSGovernment. rke-tools uses the 'cloud' entry as part of the cloud-config (https://github.com/rancher/rke-tools/blob/64c1facb4b6e5685f9ec70d0bf8074035d8acc3e/windows/cloud-provider.psm1#L20), and therefore fails with the following error message:

``` 
INFO: Waiting for servcie-sidekick to prepare the entrypoint.ps1 ... 
ERROR: The cloud 'AzureUSGovernmentCloud' is not registered. 
FATA: Failed to set 'AzureUSGovernmentCloud' as cloud type: 
INFO: Waiting for servcie-sidekick to prepare the entrypoint.ps1 ... 
ERROR: The cloud 'AzureUSGovernmentCloud' is not registered. 
FATA: Failed to set 'AzureUSGovernmentCloud' as cloud type: 
INFO: Waiting for servcie-sidekick to prepare the entrypoint.ps1 ... 
``` 

**Result:**

This error loops and prevents the Windows node from joining.

**Environment information**
- Rancher version: v2.3.3
- RKE version: local v1.0.0
- k8s version: local v1.16.3 ; cluster v1.15.7
- HA is enabled

**Comment by luthermonson** [comment]
per convo with ross/hussein we might be able to solve this with existing config

**Comment by deniseschannon** [comment]
Available with v2.3.6-rc7.

**Comment by deniseschannon** [comment]
Per the customer, this is working now.
